framework_tvb/tvb/interfaces/web/controllers/simulator_controller.py

```diff
@@ -749,6 +757,10 @@ def set_monitors(self, **data):
         form = get_form_for_monitor(type(monitor))(indexes, '', common.get_current_project().id)
         form.fill_from_trait(monitor)
 
+        if isinstance(monitor, SpatialAverage):
```

Comment:
this `if isinstance ...` is ugly. Pls find another way.
Maybe `get_form_for_monitor ` could receive an extra param (e.g. the `session_stored_simulator ` to retrieve the ConnIndex from it in each Monitor Form) ?